chromium_src/components/sync/service/sync_service_impl.cc

```diff
@@ -3,36 +3,74 @@
 #include "brave/components/brave_sync/brave_sync_prefs.h"
 #include "brave/components/sync/service/brave_sync_auth_manager.h"
 #include "brave/components/sync/service/brave_sync_stopped_reporter.h"
 #include "components/prefs/pref_service.h"
+#include "components/sync/base/command_line_switches.h"
 #include "components/sync/base/sync_util.h"
 
+#if BUILDFLAG(IS_ANDROID)
+#include "base/android/jni_android.h"
+#include "brave/build/android/jni_headers/BraveQAPreferences_jni.h"
+#endif
+
 namespace syncer {
 
 GURL BraveGetSyncServiceURL(const base::CommandLine& command_line,
                             version_info::Channel channel,
                             PrefService* prefs) {
-  // Allow group policy to override sync service URL.
-  // This has a higher priority than the --sync-url command-line param.
-  // https://github.com/brave/brave-browser/issues/20431
-  if (prefs && prefs->IsManagedPreference(brave_sync::kCustomSyncServiceUrl)) {
-    std::string value(prefs->GetString(brave_sync::kCustomSyncServiceUrl));
+// TODO: Currently, we only handle the custom sync URL via GPO/Settings Page
+// on Desktop OSes. However, we should also handle it on Android in the future.
+#if !BUILDFLAG(IS_ANDROID)
+  if (prefs) {
+    std::string value = prefs->GetString(brave_sync::kCustomSyncServiceUrl);
     if (!value.empty()) {
       GURL custom_sync_url(value);
       // Provided URL must be HTTPS.
       if (custom_sync_url.is_valid() &&
           custom_sync_url.SchemeIs(url::kHttpsScheme)) {
-        DVLOG(2) << "Sync URL specified via GPO: "
-                 << prefs->GetString(brave_sync::kCustomSyncServiceUrl);
-        return custom_sync_url;
+        // Allow group policy to override sync service URL.
+        // This has a higher priority than the --sync-url command-line param.
+        // https://github.com/brave/brave-browser/issues/20431
+        if (prefs->IsManagedPreference(brave_sync::kCustomSyncServiceUrl)) {
+          DVLOG(2) << "Sync URL specified via GPO: " << value;
+          return custom_sync_url;
+        } else {
+          // If the sync URL is not set via command line.
+          if (!command_line.HasSwitch(syncer::kSyncServiceURL) ||
+              command_line.GetSwitchValueASCII(syncer::kSyncServiceURL)
+                  .empty()) {
+            DVLOG(2) << "Sync URL specified via settings: " << value;
+            return custom_sync_url;
+          }
+        }
       } else {
-        LOG(WARNING) << "The following sync URL specified via GPO "
-                     << "is invalid: " << value;
+        LOG(WARNING) << "The following sync URL specified via GPO or settings "
+                        "page is invalid: "
+                     << value;
       }
     }
   }
+#else
+  const char kBraveSyncServiceStagingURL[] =
+      "https://sync-v2.bravesoftware.com/v2";
+  const char kProcessTypeSwitchName[] = "type";
+
+  // On Android we can detect data dir only on host process, and we cannot
```

Comment:
@jagadeshjai 
With your change no need for the workaround with check for renderer/gpu process.  Because `BraveGetSyncServiceURL` will be invoked when we will be creating the sync service, so we are at the host process and have Java initialized on Android.

You need to remove `kProcessTypeSwitchName` and the condition of `HasSwitch(kProcessTypeSwitchName)`